File: app/src/main/java/wannabit/io/cosmostaion/activities/QRcodeActivity.java
package wannabit.io.cosmostaion.activities;

import android.content.Context;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.journeyapps.barcodescanner.CaptureActivity;
import com.journeyapps.barcodescanner.CaptureManager;
import com.journeyapps.barcodescanner.DecoratedBarcodeView;

import wannabit.io.cosmostaion.R;
import wannabit.io.cosmostaion.utils.LanguageUtil;

public class QRcodeActivity extends CaptureActivity {

    Button disconnectBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        disconnectBtn = findViewById(R.id.btn_disconnect);
        disconnectBtn.setOnClickListener(v -> onBackPressed());
    }

    @Override
    protected DecoratedBarcodeView initializeContent() {
        setContentView(R.layout.activity_qrcode);
        return findViewById(R.id.zxing_barcode_scanner);
    }

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(LanguageUtil.updateResources(newBase));
    }
}

File: app/src/main/java/wannabit/io/cosmostaion/ui/qr/QrCodeActivity.kt
package wannabit.io.cosmostaion.ui.qr

import android.content.Context
import android.os.Build
import android.os.Bundle
import android.view.View
import android.view.WindowInsetsController
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.journeyapps.barcodescanner.CaptureActivity
import com.journeyapps.barcodescanner.DecoratedBarcodeView
import wannabit.io.cosmostaion.common.BaseUtils
import wannabit.io.cosmostaion.databinding.ActivityQrCodeBinding

class QrCodeActivity : CaptureActivity() {

    private lateinit var binding: ActivityQrCodeBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initFullScreen()
        binding.btnDisconnect.setOnClickListener {
            onBackPressed()
        }
    }

    override fun initializeContent(): DecoratedBarcodeView {
        binding = ActivityQrCodeBinding.inflate(layoutInflater)
        setContentView(binding.root)
        return binding.zxingBarcodeScanner
    }

    private fun initFullScreen() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            window.setDecorFitsSystemWindows(false)
            val controller = window.insetsController
            if (controller != null) {
                controller.systemBarsBehavior = WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            }
        } else {
            window.decorView.systemUiVisibility =
                (View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN or View.SYSTEM_UI_FLAG_LAYOUT_STABLE)
        }
    }

    override fun attachBaseContext(newBase: Context?) {
        super.attachBaseContext(BaseUtils.updateResources(newBase))
    }
}